addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -2255,6 +2255,9 @@ export class OdooEditor extends EventTarget {
         this.options.beforeAnyCommand();
 
         if (editorCommands[method]) {
+            if (this.powerbox.isOpen && !sel.isCollapsed) {
+                this.deleteRange();
+            }
```

Comment:
`deleteRange` is already called a few lines above; you can just add your `powebox.isOpen` check there.